cli/types/cypress.d.ts

```diff
@@ -2780,6 +2788,12 @@ declare namespace Cypress {
     /**
      * Glob pattern to determine what test files to load.
      */
+    specPattern: string | string[]
```

Comment:
Theres a note in the Notion doc regarding specPattern usage:
> specPattern may only be specified in the e2e and component objects, and is invalid at the top level of the config (will error if specified at the top level)

Should `specPattern` be defined for this type if its usage is invalid at the top level? Or is it a pain to compose it into the `e2e`/`component` config types individually?

Reply:
Top level `specPattern` should be invalid. For now, we don't have a good way to validate "only top level" or "only nested" config yet, but that will be done soon, partially in [this PR](https://github.com/cypress-io/cypress/pull/19347) and the PR that'll follow it up. I'll add you as a reviewer when this is updated; for now, to minimize noise and PR size, there isn't any additional validation implemented in this PR.